### DISCUSSION TITLE
Insights on 'Nuxt 3 and everything you use in your project are now devDependencies'

### DISCUSSION THREAD

**Discussion by mrtnvh** [nuxt/nuxt#10400] [General]
Hi everyone!

Could someone from the Nuxt team / contributers perhaps give more insights on the following

> This means that Nuxt 3 and everything you use in your project are now [devDependencies](https://nodejs.dev/learn/npm-dependencies-and-devdependencies). Nitro will take care of code-splitting and bundling everything into a portable and compact .output directory ready to be deployed to [any hosting provider ](https://nitro.unjs.io/deploy/).

From https://nuxtjs.org/announcements/nuxt3-rc/

---

My current understanding of dependencies and devDependencies is: If your user is touching a codebase that is originally (before transpilation, build, ...) from a node_module or such codebase is needed at runtime, then it's a dependency. If not then, but it's needed during build or development, then it's a devDependency.

Thank you!

**Comment by danielroe** [comment]
When building your app, any runtime dependencies are copied to `.output/server/node_modules` using https://github.com/vercel/nft. So you should never need to install dependencies in production; you just copy the `.output` directory (or use a deployment step listed on https://nitro.unjs.io/deploy).

**Comment by mrtnvh** [comment]
Hi Daniel, thank you for your response and showing me the `nft` package. Very interesting way of including external codebases in production code.

I however, cannot distil an argument on why Nuxt and all other dependencies in eg. my project, should be installed as devDependencies. Even more so, because `nft` copies and therefore directly depends on the original codebase from the dependency, enforces the argument that runtime dependencies (incl. Nuxt) should be considered `dependencies` instead of `devDependencies`

Am I missing something?

**Reply by danielroe** [reply]
It is up to you how you organise your `package.json`. Different people have different semantics for what is a dependency. As far as your Nuxt 3 project is concerned, you can put everything into devDependencies and it will continue to work, unlike with Nuxt 2.

**Reply by mrtnvh** [reply]
Thanks for clarifying. 

I understand that from a technical perspective, because the output application does not rely on the package.json in the root of the project, but on a newly created one in the output/server folder, every dependency now can be considered a devDependency.

**Reply by fanckush** [reply]
@danielroe is it still ok to put Nuxt in devDependencies in late 2024?
it looks like recent templates consider Nuxt a normal dependency

**Reply by danielroe** [reply]
It makes no difference - totally up to you.
